### ISSUE TITLE
[3.5.x-dev] Bug when saving elements that contain an asset in PostgreSQL

### ISSUE THREAD

**Issue by Marcuzz** [craftcms/cms#6287]
### Description

When saving an element that contains an asset it attempts to make a query that contains the following part:

```
WHERE ("assets"."volumeId"=':empty:')
```

This might work in MySQL but in PostgreSQL it's a type mismatch:

```
SQLSTATE[22P02]: Invalid text representation: 7 ERROR: invalid input syntax for integer: ":empty:"
```

The culprit is located here: https://github.com/craftcms/cms/blob/3.5/src/fields/Assets.php#L522

I'm not sure if there are more similar occurrences but hopefully not 😅 - I thought `:empty:` as converted to say `null` in Elements?

### Steps to reproduce

1. On the latest 3.5 beta version and using PostgreSQL as your database, create a section with an asset field
2. Make sure the asset field has multiple volumes so that it's not a "single folder"
3. Create an element in this section with an asset selected
4. You should now have a 500 internal server error

### Additional info

- Craft version: 3.5.x-dev 40a935c
- PHP version: 7.4.5
- Database driver & version: PostgreSQL 12.2
- Plugins & versions: I have used Navigation and Neo Matrix on this installation, but they shouldn't cause this issue

**Comment by brandonkelly** [comment]
This looks like the same bug that was fixed by dc1ae8b64dd21ce77cf82e1d51daecd80dc7d858. Are you sure you’re running the 3.5 commit you think you’re running?

**Comment by Marcuzz** [comment]
Yeah I'm sure, I copied the commit ref from my terminal (when running `composer update`) and cross referenced it from `package.lock` You should be able to recreate it with the same dataset I sent you from my previous issue (#6227)

**Comment by brandonkelly** [comment]
Sorry, able to reproduce. Guess I should have tested on Postgres :)

Just fixed this SQL error via 3596932cc042ccfc4efeb4d040611302f695ee2e.
